src/Text/Pandoc/Readers/HTML.hs

```diff
@@ -1301,6 +1289,17 @@ isSpace '\n' = True
 isSpace '\r' = True
 isSpace _    = False
 
+-- Utilities
+
+-- | Adjusts a url according to the document's base URL.
+canonicalizeUrl :: PandocMonad m => String -> m String
+canonicalizeUrl url = do
+  mbBaseHref <- baseHref <$> getState
```

Comment:
One problem here is that you're using `getState`, which is defined in Parsec monads, but the type sig says this works in any instance of `PandocMonad`.  Does that help?

Reply:
You probably want
```
PandocMonad m => String -> TagParser m String
```

Reply:
Thank you for the tip :smile: